src/jsSharedMain/kotlin/app/softwork/routingcompose/BrowserRouter.kt

```diff
@@ -24,7 +24,7 @@ public fun BrowserRouter(
     initPath: String,
     routeBuilder: @Composable RouteBuilder.() -> Unit
 ) {
-    BrowserRouter().route(initPath, routeBuilder)
+    Router.internalGlobalRouter = BrowserRouter().apply { route(initPath, routeBuilder) }
```

Comment:
What's the use-case/code for setting this variable? Can't the function just return the Router instead?

Reply:
It could. I've implemented it this way just to simplify the client code and to be consistent with `Router.current`. In my applications I just use `Router.global` when `Router.current` can't be called.

Reply:
What do you mean with `can't be called`? Because `Router.current` is `@Composable` or because there is no value set?

Reply:
It can't be called from any non-composable functions like effects, event handlers, view model methods or some global utility functions.